ext/sdk/resources/sdk-root/shell/src/resource-templates/ts/scaffolder.ts

```diff
@@ -0,0 +1,212 @@
+
+
+    // webpack and package.json
+    promises.push(this.fsService.writeFile(relativePath('webpack.config.js'), getWebpackContent()));
+    promises.push(this.fsService.writeFile(relativePath('package.json'), getPackageContent(resourceName))); 
+    
+    // client
+    promises.push(
+      this.fsService.mkdirp(relativePath('client'))
+        .then(() => concurrently(
+          this.fsService.writeFile(relativePath('client/client.ts'), ''),
+          this.fsService.writeFile(relativePath('client/tsconfig.json'), getTsConfigClient())
+        )),
+    );
+
+    // server
+    promises.push(
+      this.fsService.mkdirp(relativePath('server'))
+        .then(() => concurrently(
+          this.fsService.writeFile(relativePath('server/server.ts'), ''),
+          this.fsService.writeFile(relativePath('server/tsconfig.json'), getTsConfigServer())
+        )),
+    );
+
+
+    await Promise.all(promises);
+  };
+}
+
+function getTsConfigClient() {
+  return `
+{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "*": ["types/*"]
+    },
+    "outDir": "./",
+    "noImplicitAny": true,
+    "module": "commonjs",
+    "target": "es6",
+    "allowJs": true,
+    "lib": ["es2017"],
+    "types": ["@citizenfx/client", "@types/node"],
+    "moduleResolution": "node",
+    "resolveJsonModule": true,
+    "esModuleInterop": true
+  },
+  "include": ["./**/*"],
+  "exclude": ["**/node_modules"]
+}
+  `
+}
+
+
+function getTsConfigServer() {
+  return `
+  {
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "*": ["types/*"]
+    },
+    "outDir": "./",
+    "noImplicitAny": true,
+    "module": "commonjs",
+    "target": "es6",
+    "allowJs": false,
+    "lib": ["es2015", "DOM"],
+    "types": ["@citizenfx/server", "@types/node"],
+    "moduleResolution": "node",
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
+    "resolveJsonModule": true,
+    "esModuleInterop": true
+  },
+  "include": ["./**/*"],
+  "exclude": ["**/node_modules"]
+}
+  `
+}
+
+function getWebpackContent() {
+  return `
+const webpack = require('webpack');
+const path = require('path');
+const RemovePlugin = require("remove-files-webpack-plugin");
+
+const buildPath = path.resolve(process.cwd(), 'dist');
+
+const client = {
+  entry: './client/client.ts',
+
+  module: {
+    rules: [
+      {
+        test: /\.tsx?$/,
+        use: "ts-loader",
+        exclude: /node_modules/,
+      },
+    ],
+  },
+  plugins: [
+    new webpack.DefinePlugin({ "global.GENTLY": false }),
+    new RemovePlugin({
+      before: {
+        include: [path.resolve(buildPath, "client")]
+      },
+      watch: {
+        include: [path.resolve(buildPath, "client")]
+      }
+    })
+  ],
+  optimization: {
+    minimize: true
+  },
+  resolve: {
+    extensions: [".tsx", ".ts", ".js"]
+  },
+
+  output: {
+    path: path.resolve(buildPath, "client"),
+    filename: "[contenthash].client.js"
```

Comment:
Why would it need content hash?

Reply:
Just an habit, heh. We don't really need it as there are no filenames that collides.

Reply:
Seems like we need [contenthash] on server side. If you are referring to both client and server. On client it is not needed, correct.

Reply:
Why is it needed on server side?

Reply:
Without it, when a resource is changed it seemed to have the behavior of being deleted (the server side file). I kept trying for a while and see if there was any other reason for this, but could not find any. Would actually be nice if you could test it yourself, and let me know.